### ISSUE TITLE
Lint/InheritException false positive

### ISSUE THREAD

**Issue by jcoyne** [rubocop-hq/rubocop#6770]
## Expected behavior

It should not give an offense unless we are inheriting from `::Exception`

## Actual behavior

It gives an offense when we inherit from classes named `Exception` within a module (i.e. classes that are not `::Exception`.)

## Steps to reproduce the problem

```ruby
module MyModule
  class Exception < ::StandardError; end

  class AnotherError < Exception; end
end
```

## RuboCop version

```
$ bundle exec rubocop -v
0.60.0
```

**Comment by tejasbubane** [comment]
I feel this varies on Ruby's constant lookup and is out of scope of static analysis of rubocop. i.e. how will Rubocop figure out if `MyModule` has `Exception` class defined?
